### PR TITLE
Feature/lesson4 7 tables&endpoints

### PR DESCRIPTION
**Pull request by smo1in**
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by smo1in** [comment]
[Lesson4-7](https://trello.com/c/5uAnRQlh/43-lesson4-7)

**Comment by smo1in** [comment]
[Lesson4-7](https://trello.com/c/5uAnRQlh/43-lesson4-7)

**Comment by smo1in** [comment]
[Lesson4-7](https://trello.com/c/5uAnRQlh/43-lesson4-7)

File: app/code/Vendor/Voucher/Model/VoucherManagement.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    public function createVoucher($customer_id, $status_id, $voucher_code)
+    {
+        $voucher = $this->voucherModelFactory->create();
+        $voucher->setCustomerId($customer_id);
+        $voucher->setStatusId($status_id);
+        $voucher->setVoucherCode($voucher_code);
+        try {
+            $this->voucherResourceFactory->save($voucher);
+            $response = ['idVoucher' => $voucher->getId()];
+            return json_encode($response);
```
Comment (epolish) [inline app/code/Vendor/Voucher/Model/VoucherManagement.php:109]: You should never use internal php functions.

**Review by epolish — APPROVED** [review]

**Pull request merged**